contracts/peripheral/GaugeVoteRewards.sol

```diff
@@ -0,0 +1,293 @@
+    }
+
+    /* -------------------------------------------------------------------
+       Setters 
+    ------------------------------------------------------------------- */
+
+    function setDistributor(address _distributor) external onlyOwner {
+        distributor = _distributor;
+        emit SetDistributor(_distributor);
+    }
+
+    function setRewardPerEpoch(uint256 _rewardPerEpoch) external onlyOwner {
+        rewardPerEpoch = _rewardPerEpoch;
+        emit SetRewardPerEpoch(_rewardPerEpoch);
+    }
+
+    function setPoolIds(uint256[] memory _poolIds) external {
+        uint256 poolIdsLen = _poolIds.length;
+        for (uint256 i = 0; i < poolIdsLen; i++) {
+            uint256 pid = _poolIds[i];
+            IBooster.PoolInfo memory poolInfo = booster.poolInfo(pid);
+            getPoolId[poolInfo.gauge] = pid;
+        }
+    }
+
+    function setDstChainId(address[] memory _gauges, uint16[] memory _dstChainIds) external onlyOwner {
+        uint256 dstChainIdsLen = _dstChainIds.length;
+        require(dstChainIdsLen == _gauges.length, "!length");
+        for (uint256 i = 0; i < dstChainIdsLen; i++) {
+            getDstChainId[_gauges[i]] = _dstChainIds[i];
+        }
+    }
+
+    function setChildGaugeVoteRewards(uint16[] memory _dstChainIds, address[] memory _voteRewards) external onlyOwner {
+        uint256 dstChainIdsLen = _dstChainIds.length;
+        require(dstChainIdsLen == _voteRewards.length, "!length");
+        for (uint256 i = 0; i < dstChainIdsLen; i++) {
+            getChildGaugeVoteRewards[_dstChainIds[i]] = _voteRewards[i];
+        }
+    }
+
+    /* -------------------------------------------------------------------
+       View 
+    ------------------------------------------------------------------- */
+
+    function getCurrentEpoch() external view returns (uint256) {
+        return _getCurrentEpoch();
+    }
+
+    /* -------------------------------------------------------------------
+       Core 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @notice  Wraps the booster.voteGaugeWeight call to track weights for each epoch
+     *          So AURA rewards can be distributed pro rata to those pool stashes
+     * @param _gauge    Array of the gauges
+     * @param _weight   Array of the weights
+     * @return bool for success
+     */
+    function voteGaugeWeight(address[] calldata _gauge, uint256[] calldata _weight) external onlyOwner returns (bool) {
+        // Loop through each gauge and store it's weight for this epoch
+        for (uint256 i = 0; i < _gauge.length; i++) {
+            address gauge = _gauge[i];
+            uint256 weight = _weight[i];
+            // If the weight for this gauge for this epoch has already been set then revert
+            require(getAmountToSendByEpoch[_getCurrentEpoch()][gauge] == 0, "stored amountToSend!=0");
+            uint256 amountToSend = rewardPerEpoch.mul(weight).div(TOTAL_WEIGHT_PER_EPOCH);
```

Comment:
Some gauges don't have deposits but might have votes (veLIT, veBAL) so we need to account for their votes here. We should track the weights and a total weight and then calculate the amountToSend when we call process.